solidity/contracts/Gravity.sol

```diff
@@ -58,6 +58,17 @@ struct ValsetArgs {
 	address rewardToken;
 }
 
+// This is being used purely to avoid stack too deep errors
+struct PaymentArgs {
+	// Arrays containing the batch data
+	uint256[] amounts;
```

Comment:
Here you could use a nested struct, and thereby avoid having to check that all of those arrays are the same length:

```
struct TransactionArgs {
     uint256 amount;
     address destination;
     uint256 fees;
}

struct PaymentArgs {
      TransactionArgs[] transactionArgs;
      address feePaymentAddress;
}
```

I'd guess this is more gas efficient since you can avoid some checks and have only one array to index.

Reply:
thanks for the comment,

the problem is that I would have to rebuild the array anyways as they are used for the input to compute the hash root
https://github.com/crypto-org-chain/gravity-bridge/pull/143/files#diff-539af61444a15e77d6b01973fbd4876d6300fbe84dec2277fbb75d7e3e29de6fR452

is your comment still valid?

Reply:
Oh yes, thats true. For this to make sense validators would have to sign a different hash based on the new datastructure, which is probably outside of the scope of this PR.